### ISSUE TITLE
Problem with the channelUpdate event

### ISSUE THREAD

**Issue by HamoodiHajjiri** [discordjs/discord.js#7916]
### Which package is this bug report for?

discord.js

### Issue description

I was willing to use the `GuildChannel.position` property for both the `oldChannel` and the `newChannel` parameters, however, I've realised that both values return the same number.

### Code sample

```typescript
// Checking if the channel position has changed. If it has, it will return. 
if (oldChannel.position !== newChannel.position) return;
```

### Package version

V13.6.0

### Node.js version

V18.0.0

### Operating system

Windows 11 Pro

### Priority this issue should have

Low (slightly annoying)

### Which partials do you have configured?

User, Channel, GuildMember, Message, Reaction, GuildScheduledEvent, ThreadMember

### Which gateway intents are you subscribing to?

Guilds, GuildMembers, GuildBans, GuildEmojisAndStickers, GuildIntegrations, GuildWebhooks, GuildInvites, GuildVoiceStates, GuildPresences, GuildMessages, GuildMessageReactions, GuildMessageTyping, DirectMessages, DirectMessageReactions, DirectMessageTyping, GuildScheduledEvents

**Comment by almeidx** [comment]
This is because `position` is a getter, which calls the `Guild#_sortedChannels()` method under the hood, which sorts all the channels (from `<Client>.channels.cache`)
https://github.com/discordjs/discord.js/blob/e3cbd45e7de069432818148b9dedda03f630f9f7/src/structures/GuildChannel.js#L146-L149

https://github.com/discordjs/discord.js/blob/e3cbd45e7de069432818148b9dedda03f630f9f7/src/structures/Guild.js#L1409-L1420

Because of this, both `oldChannel.position` and `newChannel.position` will always return the (same) most up-to-date position

**Comment by almeidx** [comment]
I suppose one way of working around this issue is to manually set the `position` on the clone to the value of the getter, before patching the channel
https://github.com/discordjs/discord.js/blob/e3cbd45e7de069432818148b9dedda03f630f9f7/src/structures/Base.js#L28-L32

Something like:
```js
 _update(data) { 
   const clone = this._clone(); 
   Reflect.defineProperty(clone, 'position', { value: clone.position });
   this._patch(data); 
   return clone; 
 } 
```
*this would be done on the GuildChannel class, not Base*

However, this introduces some overhead on every channel update, and will still be prone to race conditions when multiple channels are updated at the same time

**Comment by HamoodiHajjiri** [comment]
I suppose a way around this is to use `.rawPosition` to not have both return the (same) most up-to-date position, correct?

**Comment by pranshu05** [comment]
> I suppose one way of working around this issue is to manually set the `position` on the clone to the value of the getter, before patching the channel
> 
> https://github.com/discordjs/discord.js/blob/e3cbd45e7de069432818148b9dedda03f630f9f7/src/structures/Base.js#L28-L32
> 
> Something like:
> 
> ```js
>  _update(data) { 
>    const clone = this._clone(); 
>    Reflect.defineProperty(clone, 'position', { value: clone.position });
>    this._patch(data); 
>    return clone; 
>  } 
> ```
> 
> _this would be done on the GuildChannel class, not Base_
> 
> However, this introduces some overhead on every channel update, and will still be prone to race conditions when multiple channels are updated at the same time

Yes... Cloning values works

**Comment by JMTK** [comment]
Can you confirm if this is fixed after this is now in main? https://github.com/discordjs/discord.js/pull/9497

**Comment by almostSouji** [comment]
resolved as of 09b0382c458ed0f3df5b8fa778c504b3e658ac49
if you can still reproduce, please open a new issue with sufficient details
